File: 1496-path-crossing/1496-path-crossing.java
class Solution {
    public boolean isPathCrossing(String path) {
        int x = 0;
        int y = 0;
        Set<String> curr =new HashSet<String>();
        curr.add("0,0");

        for(char ele : path.toCharArray()){
            if(ele == 'N') x++;
            else if(ele == 'S') x--;
            else if(ele == 'E') y++;
            else if(ele == 'W') y--; 

            String position = x+","+y;
            if(curr.contains(position)) return true;
            curr.add(position);
        }
        return false;
    }
}

File: 1496-path-crossing/1496-path-crossing.js
/**
 * @param {string} path
 * @return {boolean}
 */
function isPathCrossing(path) {
    let x = 0, y = 0;
    let curr = new Set();
    curr.add("0,0");

    for (let position of path) {
        if (position === 'E') x++;
         else if (position === 'W') x--;
         else if (position === 'N') y++;
         else if (position === 'S') y--;
        
        let currentPos = `${x},${y}`;
        if (curr.has(currentPos)) return true;
        curr.add(currentPos);
    }
    return false;
};
